fix(rdb): new troubleshooting pages (#4315)
scaleway/docs-content: pages/managed-databases-for-postgresql-and-mysql/troubleshooting/database-instance-connectivity-issues.mdx

---
meta:
  title: Dealing with Database Instance connectivity issues
  description: Troubleshoot Database Instance connectivity issues for Managed Databases for MySQL and PostgreSQL.
content:
  h1: Dealing with Database Instance connectivity issues
  paragraph: Troubleshoot Database Instance connectivity issues for Managed Databases for MySQL and PostgreSQL.
tags: disk-full databases
dates:
  validation: 2025-01-29
  posted: 2025-01-29
categories:
  - managed-databases
  - postgresql-and-mysql
---

## Public access

### Problem

I cannot connect to my Database Instance through the public network.

### Possible causes

- The Database Instance TLS certificate is outdated
- Your allowed IPs are not properly configured
- You have reached the maximum number of connections

> - The Database Instance experiences instabilities due to a high load

Reviewer: ```suggestion
- The Database Instance is experiencing instabilities due to a high load
```

### Solution

You can check the following points to identify and act on what might be causing the issue:

> 1. If TLS is enabled on the Database Instance, make sure the certificate is up-to-date, and that your client application is properly configured.

Reviewer: ```suggestion
1. If TLS is enabled on the Database Instance, make sure the certificate is up to date, and that your client application is properly configured.
```

2. Make sure your [ACLs](/managed-databases-for-postgresql-and-mysql/concepts/#allowed-ips) are [properly configured](/managed-databases-for-postgresql-and-mysql/how-to/manage-allowed-ip-addresses).

> 3. Make sure your Database Instance does not reach the maximum number of connections stipulated in the advanced settings of the Database Instance. You can monitor the number of connections on [Cockpit](/managed-databases-for-postgresql-and-mysql/how-to/monitor-databases-cockpit). You can also [adjust the advances settings](/managed-databases-for-postgresql-and-mysql/how-to/configure-advanced-settings) to accept a higher number of connections.

Reviewer: ```suggestion
3. Make sure your Database Instance has not reached the maximum number of connections stipulated in the advanced settings of the Database Instance. You can monitor the number of connections on [Cockpit](/managed-databases-for-postgresql-and-mysql/how-to/monitor-databases-cockpit). You can also [adjust the advances settings](/managed-databases-for-postgresql-and-mysql/how-to/configure-advanced-settings) to accept a higher number of connections.
```

4. [Monitor other usage metrics on Cockpit](/managed-databases-for-postgresql-and-mysql/how-to/monitor-databases-cockpit) to:

    - Check if the memory usage is nominal. High memory usage could trigger OOM Kills and cause disconnection
    - Check Database Instance logs and look for anything else that could explain the issue

## Private access

### Problem

I cannot connect to my Database Instance through a Private Network.

> ### Possible causes

Reviewer 2: we can also mention TLS certificate issue and instabilities due to high load? Same as for public connection issue

Author: I had, but @bastienwirtz asked to remove it, what do we keep?

Reviewer 3: My bad, my point was only regarding the ACL / allowed ip that does not applied on private network. We can keep the TLS issue. Regarding the load it's already mentioned line 52.

- Client network issue
- You have reached the maximum number of connections

> - You have a high memory usage

Reviewer 3: ```suggestion
- The Database Instance is experiencing instabilities due to a high load
```

> ### Solution

Reviewer 2: we could also add or refer to the public connectivity solutions for TLS certificate, maximum connections & instabilities due to high load?

> - Client network issue
> - You have reached the maximum number of connections
> - The Database Instance is experiencing instabilities due to a high load

Author: ```suggestion
- The Database Instance TLS certificate is outdated
- Client network issue
- You have reached the maximum number of connections
- The Database Instance is experiencing instabilities due to a high load
```

You can carry out the following actions:

> 1. Check if you are you able to connect to the Database Instance from the public endpoint, if one is available.

Reviewer 2: that would be an information to add in the ticket no? cc @bastienwirtz 

It doesn't really give a solution to the user?

content:
  h1: Dealing with Database Instance connectivity issues
  paragraph: Troubleshoot Database Instance connectivity issues for Managed Databases for MySQL and PostgreSQL.
tags: disk-full databases
dates:
  validation: 2025-01-29
  posted: 2025-01-29
categories:
  - managed-databases
  - postgresql-and-mysql
---

## Public access

### Problem

I cannot connect to my Database Instance through the public network.

### Possible causes

- The Database Instance TLS certificate is outdated
- Your allowed IPs are not properly configured
- You have reached the maximum number of connections
- The Database Instance is experiencing instabilities due to a high load

### Solution

You can check the following points to identify and act on what might be causing the issue:

1. If TLS is enabled on the Database Instance, make sure the certificate is up to date, and that your client application is properly configured.
2. Make sure your [ACLs](/managed-databases-for-postgresql-and-mysql/concepts/#allowed-ips) are [properly configured](/managed-databases-for-postgresql-and-mysql/how-to/manage-allowed-ip-addresses).

> 3. Make sure your Database Instance has not reached the maximum number of connections stipulated in the advanced settings of the Database Instance. You can monitor the number of connections on [Cockpit](/managed-databases-for-postgresql-and-mysql/how-to/monitor-databases-cockpit). You can also [adjust the advances settings](/managed-databases-for-postgresql-and-mysql/how-to/configure-advanced-settings) to accept a higher number of connections.

Reviewer 4: ```suggestion
3. Make sure your Database Instance has not reached the maximum number of connections stipulated in the advanced settings of the Database Instance. You can monitor the number of connections on [Cockpit](/managed-databases-for-postgresql-and-mysql/how-to/monitor-databases-cockpit). You can also [adjust the advanced settings](/managed-databases-for-postgresql-and-mysql/how-to/configure-advanced-settings) to accept a higher number of connections.
```

4. [Monitor other usage metrics on Cockpit](/managed-databases-for-postgresql-and-mysql/how-to/monitor-databases-cockpit) to:

    - Check if the memory usage is nominal. High memory usage could trigger OOM Kills and cause disconnection
    - Check Database Instance logs and look for anything else that could explain the issue

## Private access

### Problem

I cannot connect to my Database Instance through a Private Network.

### Possible causes

- The Database Instance TLS certificate is outdated
- Client network issue
- You have reached the maximum number of connections
- The Database Instance is experiencing instabilities due to a high load

### Solution

You can carry out the following actions:

> 1. Try to connect to the Database Instance from the public endpoint, if one is available, to identify the possible origin (network or instance). This information can help you or the support team to identify the issue.

Reviewer 5: ```suggestion
1. Try to connect to the Database Instance from the public endpoint, if one is available, to identify the possible origin (network or Instance). This information can help you or the support team to identify the issue.
```

not sure though but normally if Instance refers to the product it should have an uppercase I
